x-pack/plugins/dashboard_mode/server/dashboard_mode_request_interceptor.js

```diff
@@ -25,9 +27,27 @@ export function createDashboardModeRequestInterceptor(dashboardViewerApp) {
     type: 'onPostAuth',
     async method(request, h) {
       const { auth, url } = request;
+      const user = auth.credentials;
+      const roles = user ? user.roles : [];
+
+      if (!user) {
+        return h.continue;
+      }
+
       const isAppRequest = url.path.startsWith('/app/');
 
-      if (isAppRequest && auth.credentials.scope && auth.credentials.scope.includes(AUTH_SCOPE_DASHBORD_ONLY_MODE)) {
+      const uiSettings = request.getUiSettingsService();
+      const dashboardOnlyModeRoles = await uiSettings.get(CONFIG_DASHBOARD_ONLY_MODE_ROLES);
+
+      if (!isAppRequest || !dashboardOnlyModeRoles || !roles || roles.length === 0) {
```

Comment:
__optional nit:__ I'd say the perf impact of `find` on `roles` is negligible comparing to async `uiSettings.get` call, so maybe there is no need to have this additional return path (and assuming we'll do roles check in the `if` above) and always treat this advanced setting a defined array.
```ts
const dashboardOnlyModeRoles = await uiSettings.get(CONFIG_DASHBOARD_ONLY_MODE_ROLES) || [];
```

__nit:__ also can you please leave a comment in code explaining why we always call `uiSettings.get`, so that we don't forget that?